VimeoNetworking/Sources/AuthenticationController.swift

```diff
@@ -564,9 +571,7 @@ final public class AuthenticationController
         {
             try self.setClientAccount(with: account, shouldClearCache: true)
             
-            let accountType: AccountStore.AccountType = (account.user != nil) ? .User : .ClientCredentials
-            
-            try self.accountStore.saveAccount(account, type: accountType)
+            try self.saveAccount(account)
```

Comment:
I can give it a try, but might potentially be a large refactor. It'd require us to move all methods that refresh the user and isolate it into AuthenticationController so that we have one place that updates the user.